proof/refine/ARM/Tcb_R.thy

```diff
@@ -3475,23 +3475,25 @@ lemma cap_CNode_case_throw:
      = (doE unlessE (isCNodeCap cap) (throw x); m odE)"
   by (cases cap, simp_all add: isCap_simps unlessE_def)
 
-lemma decode_set_space_corres:
+lemma decodeCVSpace_corres:
   notes if_cong [cong]
   shows
- "\<lbrakk> cap_relation cap cap'; list_all2 (\<lambda>(c, sl) (c', sl'). cap_relation c c' \<and> sl' = cte_map sl) extras extras';
-      is_thread_cap cap \<rbrakk> \<Longrightarrow>
-  corres (ser \<oplus> tcbinv_relation)
-         (invs and valid_cap cap and (\<lambda>s. \<forall>x \<in> set extras. cte_at (snd x) s))
-         (invs' and valid_cap' cap'  and (\<lambda>s. \<forall>x \<in> set extras'. cte_at' (snd x) s))
-      (decode_set_space args cap slot extras)
-      (decodeSetSpace args cap' (cte_map slot) extras')"
-  apply (simp    add: decode_set_space_def decodeSetSpace_def
+ "\<lbrakk>cap_relation cap cap';
+   list_all2 (\<lambda>(c, sl) (c', sl'). cap_relation c c' \<and> sl' = cte_map sl) extras extras';
+   is_thread_cap cap\<rbrakk> \<Longrightarrow>
+  corres (ser \<oplus> (\<lambda>abs_inv conc_inv. tcbinv_relation abs_inv conc_inv
+                                     \<and> Invocations_A.is_ThreadControlCaps abs_inv
+                                     \<and> is_ThreadControlCaps conc_inv))
```

Comment:
I might be wrong, but I think it's a bit unusual to have these sort of conditions in the return relation of a corres statement. Would it be possible to prove them as post conditions of a separate hoare triple?
This is pretty minor, so I'm fine with leaving it as is if having them here is more useful.